ambry-frontend/src/main/java/com.github.ambry.frontend/FrontendMetrics.java

```diff
@@ -392,4 +386,50 @@ public FrontendMetrics(MetricRegistry metricRegistry) {
     putWithAccountAndContainerHeaderRate = metricRegistry.meter(
         MetricRegistry.name(AmbryBlobStorageService.class, "PutWithAccountAndContainerHeaderRate"));
   }
+
+  /**
+   * Class to hold different {@link RestRequestMetrics} for plain text, ssl, with and without encryption
+   */
+  class RestRequestMetricsGroup {
+    private final RestRequestMetrics blobMetrics;
+    private final RestRequestMetrics blobSslMetrics;
+    private final RestRequestMetrics blobEncryptedMetrics;
+    private final RestRequestMetrics blobSslEncryptedMetrics;
+
+    /**
+     * Instantiates {@link RestRequestMetricsGroup} for the given requestType and resource
+     * @param requestType refers to the type of request (Head/Get/Post)
+     * @param resource refers to the resource type (Blob/BlobInfo/UserMetadata)
+     */
+    RestRequestMetricsGroup(String requestType, String resource) {
+      blobMetrics = new RestRequestMetrics(AmbryBlobStorageService.class, requestType + resource, metricRegistry);
```

Comment:
very minor: is the "blob" prefix on all of these variables required?

Reply:
I thought about that too. Then, non encryted and non ssl can't be named just "metrics" and might have to go with restReqeustMetrics and inturn will bloat every other metrics variable. Hence went with this. Let me know what is your take here.